xmtp_mls/migrations/2024-05-11-004236_cache_association_state/up.sql

```diff
@@ -0,0 +1,6 @@
+CREATE TABLE association_state (
+    "inbox_id" TEXT NOT NULL,
+    "sequence_id" BIGINT NOT NULL,
+    "state" BLOB NOT NULL,
```

Comment:
In the future we could consider removing unused cached state. I'm not sure if it's safe to assume that only the latest sequence_id needs to be cached. We could consider adding the client time here, and expiring old entries

Reply:
We almost need some sort of reference counter. It's only desirable to delete when no conversations reference that sequence ID.

Reply:
Is this correct - If no new members of an inbox_id get added to the group, then using the association_state with old sequence id of that inbox_id is fine.

Reply:
It's valid for a group to stay on an older sequence_id. But we do try and get groups up-to-date.

Group members periodically do checks to see if any members have newer identity updates, and will publish the new `sequence_id` to the group if they find any. This happens any time you add or remove members, as well as periodically when calling sync.

Reply:
I'm wondering if we can do some kind of combined heuristic - we clean up caches if they are more than a day old and a newer sequence ID is cached. If so, I can address in a follow-up PR